src/Microsoft.VisualStudio.Threading.Analyzers/VSTHRD001UseSwitchToMainThreadAsyncAnalyzer.cs

```diff
@@ -32,26 +32,41 @@ public override void Initialize(AnalysisContext context)
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
 
-            context.RegisterSyntaxNodeAction(Utils.DebuggableWrapper(this.AnalyzeInvocation), SyntaxKind.InvocationExpression);
+            context.RegisterCompilationStartAction(compilationStartContext =>
+            {
+                var legacyThreadSwitchingMembers = CommonInterest.ReadMethods(compilationStartContext.Options, CommonInterest.FileNamePatternForLegacyThreadSwitchingMembers, compilationStartContext.CancellationToken).ToImmutableArray();
+                var analyzer = new Analyzer(legacyThreadSwitchingMembers);
+                compilationStartContext.RegisterSyntaxNodeAction(Utils.DebuggableWrapper(analyzer.AnalyzeInvocation), SyntaxKind.InvocationExpression);
+            });
         }
 
-        private void AnalyzeInvocation(SyntaxNodeAnalysisContext context)
+        private sealed class Analyzer
```

Comment:
nit: IMO `sealed` is a waste of space and reading time. It's private so there's no reason to limit extensibility. And if it's a micro-optimization you're going for, I'm pretty sure it's not worth it. In the meantime it leads me and perhaps others to scratch my head wondering why you went to the trouble.